FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJob.cs

```diff
@@ -393,12 +385,28 @@ private async Task<JObject> GetPatientResource(string patientId, CancellationTok
             BaseSearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
-            List<KeyValuePair<string, string>> sharedQueryParameters = new List<KeyValuePair<string, string>>(searchOptions.QueryParameters);
-
             foreach (TypeFilter typeFilter in _typeFilters)
             {
+                if (_inputData.SplitParameters != null)
+                {
+                    if (!_inputData.SplitParameters.ContainsKey(typeFilter.ResourceType))
+                    {
+                        continue;
+                    }
+
+                    searchOptions.QueryParameters.Add(new KeyValuePair<string, string>(
+                            FhirApiConstants.LastUpdatedKey,
+                            $"lt{_inputData.SplitParameters[typeFilter.ResourceType].DataEndTime.ToInstantString()}"));
+
+                    if (_inputData.SplitParameters[typeFilter.ResourceType].DataStartTime != null)
+                    {
+                        searchOptions.QueryParameters.Add(new KeyValuePair<string, string>(
+                            FhirApiConstants.LastUpdatedKey,
+                            $"ge{((DateTimeOffset)_inputData.SplitParameters[typeFilter.ResourceType].DataStartTime).ToInstantString()}"));
+                    }
+                }
+
                 searchOptions.ResourceType = typeFilter.ResourceType;
-                searchOptions.QueryParameters = new List<KeyValuePair<string, string>>(sharedQueryParameters);
                 foreach (KeyValuePair<string, string> parameter in typeFilter.Parameters)
                 {
                     searchOptions.QueryParameters.Add(parameter);
```

Comment:
seems the `searchOptions.QueryParameters` will include the parameters of previous typefilter.

Reply:
Thanks, fixed it!